modules/ip6/control/address.c

```diff
@@ -212,8 +217,11 @@ static struct api_out addr6_del(const void *request, void ** /*response*/) {
 	if ((addrs = ip6_addr_get_all(req->addr.iface_id)) == NULL)
 		return api_out(errno, 0);
 
+	scoped = req->addr.addr.ip;
+	ip6_addr_linklocal_scope(&scoped, req->addr.iface_id);
```

Comment:
I think it is safe to scope the address present in the request. no need for a local variable.